File: back/src/main/java/com/skytrix/model/dto/deck/CreateDeckDTO.java
package com.skytrix.model.dto.deck;

import jakarta.validation.constraints.Size;

import java.util.List;

import lombok.Data;

@Data
public class CreateDeckDTO {
    private Long id;

    private String name;

    @Size(max = 3)
    private List<EntityIndexDTO> imageIds;

    @Size(max = 60)
    private List<EntityIndexDTO> mainIds;

    @Size(max = 15)
    private List<EntityIndexDTO> extraIds;

    @Size(max = 15)
    private List<EntityIndexDTO> sideIds;
}

File: front/src/app/core/model/dto/create-deck-dto.ts
import { Deck } from '../deck';

export class CreateDeckDTO {
  id: number | undefined;
  name: string;
  imageIds: Array<CardIndexDTO>;
  mainIds: Array<CardIndexDTO>;
  extraIds: Array<CardIndexDTO>;
  sideIds: Array<CardIndexDTO>;

  constructor(deck: Deck) {
    this.id = deck.id;
    this.name = deck.name;
    this.imageIds = deck.cleanSlotsAndMap(deck.images, true);
    this.mainIds = deck.cleanSlotsAndMap(deck.mainDeck);
    this.extraIds = deck.cleanSlotsAndMap(deck.extraDeck);
    this.sideIds = deck.cleanSlotsAndMap(deck.sideDeck);
  }
}

export class CardIndexDTO {
  id: number;
  index: number;

  constructor(id: number, index: number) {
    this.id = id;
    this.index = index;
  }
}
